topostats/tracing/dnatracing.py

```diff
@@ -314,101 +315,65 @@ def get_splined_traces(self):
         This function actually calculates the average of several splines which is important for getting a good fit on
         the lower res data"""
 
-        step_size = int(7e-9 / (self.pixel_to_nm_scaling))  # 3 nm step size
-        interp_step = int(1e-10 / self.pixel_to_nm_scaling)
-        # Lets see if we just got with the pixel_to_nm_scaling
-        # step_size = self.pixel_to_nm_scaling
-        # interp_step = self.pixel_to_nm_scaling
-
-        self.splining_success = True
-        nbr = len(self.fitted_trace[:, 0])
-
-        # Hard to believe but some traces have less than 4 coordinates in total
-        if len(self.fitted_trace[:, 1]) < 4:
-            self.splined_trace = self.fitted_trace
-            # continue
-
-        # The degree of spline fit used is 3 so there cannot be less than 3 points in the splined trace
-        while nbr / step_size < 4:
-            if step_size <= 1:
-                step_size = 1
+        # Fitted traces are Nx2 numpy arrays of coordinates
+        # All self references are here for easy turning into static method if wanted, also type hints and short documentation
+        fitted_trace = self.fitted_trace  # : np.ndarray - boolean 2d numpy array of fitted traces to spline
+        step_size_m = self.step_size_m  # : float - the step size for the splines to skip pixels in the fitted trace
+        pixel_to_nm_scaling = self.pixel_to_nm_scaling  # : float - pixel to nanometre scaling factor for the image
+        mol_is_circular = self.mol_is_circular  # : mol_is_circular - whether or not the molecule is classed as circular
+        n_grain = self.n_grain  # : int - the grain index (for logging purposes)
+
+        # Calculate the step size in pixels
+        step_size_px = int(step_size_m / pixel_to_nm_scaling)
```

Comment:
Apologies I just noticed the actual splining func takes those instead. But what happens if in a scan our `px_2_nm_scaling` > 2*`step_size_nm` = 14nm 
-> will the int force a 0 and thus not spline any larger images regardless of what is in them?

Reply:
Yes, `int` would cause this to be zero for images that have a low `step_size_m` and a high number of nanometres per pixel (low resolution images). 

I could be wrong, but I think for the default case of 7nm for `step_size_m`, the resolution would need to be worse than 7nm per pixel before this would be calculated as zero.

Regardless, I've changed this so that `step_size_px` is always at least 1 to fix this potentiality.